subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy

```diff
@@ -302,4 +304,44 @@ class TaskUpToDateIntegrationTest extends AbstractIntegrationSpec {
         then:
         skipped(':myTask')
     }
+
+    @Unroll
+    @Issue("https://github.com/gradle/gradle/issues/4204")
+    def "changing path of empty root directory makes task out of date for #inputAnnotation"() {
+        buildFile << """
+            class MyTask extends DefaultTask {
+                @${inputAnnotation}
+                File input
+                @OutputFile
+                File output
+                
+                @TaskAction
+                void doStuff() {
+                    output.text = input.list().join('\\n')
+                }
+            }           
+            
+            File inputDir = file("inputDir" + property("select"))
```

Comment:
Nit: I would have an `inputDir` property instead that would make this simpler, but this is really a nit.